### PR TITLE
Add support for BPF_PROG_TYPE_CGROUP_SYSCTL

### PR DESCRIPTION
**Pull request by nak3**
This patch adds support for `BPF_PROG_TYPE_CGROUP_SYSCTL`.

close https://github.com/aya-rs/aya/issues/217

### CODE REVIEW

**Comment by nak3** [comment]
https://github.com/nak3/aya-showcase/tree/main/cgroup-sysctl is the demonstration for this PR.

File: aya/src/obj/mod.rs [hunk 1]
```diff
@@ -222,6 +224,7 @@ impl FromStr for ProgramSection {
                     })
                 }
             },
+            "cgroup/sysctl" => CgroupSysctl { name },
```
Comment (NoneTirex) [inline aya/src/obj/mod.rs:227]: As I see for another mappings (cgroup/skb, cgroup => skb), I think the mapping for "cgroup => sysctl" should also be implemented (for now is only cgroup/sysctl)

Comment (nak3) [reply aya/src/obj/mod.rs]: Yes, makes sense. Updated. Thank you!

File: aya/src/programs/cgroup_sysctl.rs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+use crate::{
+    generated::{bpf_attach_type::BPF_CGROUP_SYSCTL, bpf_prog_type::BPF_PROG_TYPE_CGROUP_SYSCTL},
+    programs::{
+        define_link_wrapper, load_program, FdLink, Link, OwnedLink, ProgAttachLink, ProgramData,
+        ProgramError,
+    },
+    sys::{bpf_link_create, bpf_prog_attach, kernel_version},
+};
+
+/// A program used to hook for sysctl.
```
Comment (alessandrod) [inline aya/src/programs/cgroup_sysctl.rs:15]: A program used to watch for sysctl changes.
[add empty line before next paragraph]

File: aya/src/programs/cgroup_sysctl.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+/// [`CgroupSysctl`] programs can be attached to a cgroup and will be called every
+/// time a process inside that cgroup tries to read from or write to sysctl knob in proc.
```
Comment (alessandrod) [inline aya/src/programs/cgroup_sysctl.rs:17]: to a sysctl knob (missing a)

File: aya/src/programs/cgroup_sysctl.rs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+///
+///
+/// [cgroup]: https://man7.org/linux/man-pages/man7/cgroups.7.html
```
Comment (alessandrod) [inline aya/src/programs/cgroup_sysctl.rs:20]: I don't think this is referenced by anything?

**Review by alessandrod — CHANGES_REQUESTED** [review]
Thanks a lot, looks great! Just a couple of doc nits.

File: aya/src/programs/cgroup_sysctl.rs (hunk 4 above)
Comment (nak3) [reply aya/src/programs/cgroup_sysctl.rs]: You are correct. Removed it.

**Comment by nak3** [comment]
Thank you! Updated.

**Review by alessandrod — APPROVED** [review]

**Pull request merged**